crates/ruff_linter/src/rules/flake8_comprehensions/rules/unnecessary_comprehension_in_call.rs

```diff
@@ -65,18 +66,23 @@ use crate::rules::flake8_comprehensions::fixes;
 ///
 /// [preview]: https://docs.astral.sh/ruff/preview/
 #[violation]
-pub struct UnnecessaryComprehensionInCall;
+pub struct UnnecessaryComprehensionInCall {
+    comprehension_kind: ComprehensionKind,
+}
 
 impl Violation for UnnecessaryComprehensionInCall {
     const FIX_AVAILABILITY: FixAvailability = FixAvailability::Sometimes;
 
     #[derive_message_formats]
     fn message(&self) -> String {
-        format!("Unnecessary list comprehension")
+        match self.comprehension_kind {
+            ComprehensionKind::List => format!("Unnecessary list comprehension"),
+            ComprehensionKind::Set => format!("Unnecessary set comprehension"),
```

Comment:
I changed this to a manual match rather than using `Display` for the (bad) reason that we use the first `format!` call here to show the example error message in the docs, and "Unnecessary list comprehension" is better than "Unnecessary {} comprehension".